src/Adapter/Feature/CommandHandler/EditFeatureHandler.php

```diff
@@ -26,48 +26,46 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Feature\CommandHandler;
 
-use Feature;
 use PrestaShop\PrestaShop\Adapter\Domain\AbstractObjectModelHandler;
+use PrestaShop\PrestaShop\Adapter\Feature\Repository\FeatureRepository;
 use PrestaShop\PrestaShop\Core\Domain\Feature\Command\EditFeatureCommand;
 use PrestaShop\PrestaShop\Core\Domain\Feature\CommandHandler\EditFeatureHandlerInterface;
-use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\CannotEditFeatureException;
-use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\FeatureConstraintException;
-use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\FeatureNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
 
 /**
  * Handles feature editing.
  */
-final class EditFeatureHandler extends AbstractObjectModelHandler implements EditFeatureHandlerInterface
+class EditFeatureHandler extends AbstractObjectModelHandler implements EditFeatureHandlerInterface
 {
+    /**
+     * @var FeatureRepository
+     */
+    private $featureRepository;
+
+    public function __construct(
+        FeatureRepository $featureRepository
+    ) {
+        $this->featureRepository = $featureRepository;
+    }
+
     /**
      * {@inheritdoc}
      */
-    public function handle(EditFeatureCommand $command)
+    public function handle(EditFeatureCommand $command): void
     {
-        $feature = new Feature($command->getFeatureId()->getValue());
-
-        if (empty($feature->id)) {
-            throw new FeatureNotFoundException('Feature could not be found.');
-        }
+        $feature = $this->featureRepository->get($command->getFeatureId());
 
         if (null !== $command->getLocalizedNames()) {
             $feature->name = $command->getLocalizedNames();
         }
 
-        if (false === $feature->validateFields(false)) {
-            throw new FeatureConstraintException('Invalid data when updating feature');
-        }
-
-        if (false === $feature->validateFieldsLang(false)) {
-            throw new FeatureConstraintException('Invalid data when updating feature', FeatureConstraintException::INVALID_NAME);
-        }
-
-        if (false === $feature->update()) {
-            throw new CannotEditFeatureException(sprintf('Failed to edit Feature with id "%s".', $feature->id));
-        }
+        $this->featureRepository->update($feature);
 
+        // ObjectModel::update doesn't seem to remove unassociated shops, so we must always update them manually afterwards
```

Comment:
Did you try using the `AbstractMultiShopObjectModelRepository::updateObjectModelForShops` ? It would be ideal if this was integrated in the repository update method automatically, I suggested something similar to @JevgenijVisockij the `$shopIds` should be provided as a second parameter of the update method, this way we also force always specifying the shop association (but it means it must be mandatory in the update command 🤔)

Considering how it's used in the form data handler it could be possible